src/VisualStudio/Core/Def/Implementation/AbstractEditorFactory.cs

```diff
@@ -413,4 +415,70 @@ static Project AddEditorConfigFile(Project project, string folder, out bool foun
             return project.AddAnalyzerConfigDocument(EditorConfigFileName, text, filePath: editorConfigFile).Project;
         }
     }
+
+    public int ChooseEditorFactory(
+        string pszMkDocument,
+        IVsHierarchy pHier,
+        uint itemid,
+        IntPtr punkDocDataExisting,
+        ref Guid rguidLogicalView,
+        out Guid pguidEditorTypeActual,
+        out Guid pguidLogicalViewActual)
+    {
+        // Use WinForms Editor Factory directly if view type is Designer.
+        if (IsWinFormsDesignerAsyncLoadFeatureEnabled() && VSConstants.LOGVIEWID_Designer.CompareTo(rguidLogicalView) == 0)
```

Comment:
```suggestion
        if (IsWinFormsDesignerAsyncLoadFeatureEnabled() && rguidLogicalView == VSConstants.LOGVIEWID_Designer)
```

Unless there something really subtle going on here?

Reply:
(we do the same comparisons in other parts of this file so if this is bad, I have deeper concerns!)